src/pages/UserProfile.tsx

```diff
@@ -0,0 +1,186 @@
+    fileInput.addEventListener('change', (e: Event) => handlePhotoChange(e));
+    fileInput.click();
+  };
+
+  const handlePhotoChange = (e: Event) => {
+    if (
+      e.target &&
+      e.target instanceof HTMLInputElement &&
+      e.target.files &&
+      e.target.files[0]
+    ) {
+      const reader = new FileReader();
+      reader.onload = (event) => {
+        const result = (event.target as FileReader)?.result as string;
+        formik.setFieldValue('profilePhoto', result);
+      };
+      reader.readAsDataURL(e.target.files[0]);
+    }
+  };
+
+  
+
+  return (
+    <div className="container-profile">
+      <form className="form" onSubmit={formik.handleSubmit}>
+        {userProfile ? (
+          <div className="profile-width">
+            <h1 className="heading--h1">Personal information</h1>
+            {isEditing ? (
+              <img
+                onClick={handleAddPhoto}
+                className="photo-upload"
+                src={formik.values.profilePhoto || '' || photoUpload}
+                alt="profilePhoto"
+              />
+            ) : (
+              <img
+                onClick={handleEditClick}
+                src={'' || photoUpload || formik.values.profilePhoto}
+                alt="profilePhoto"
+                className="photo-upload"
+              />
+            )}
+            <label className="label">
+              <span className="color-gray">Full name</span>
+              {isEditing ? (
+                <input
+                  type="text"
+                  {...formik.getFieldProps('username')}
+                  required
+                  className="editing-input"
+                />
+              ) : (
+                <span>{userProfile.username}</span>
+              )}
+            </label>
+            <label className="label">
+              <span className="color-gray">Email</span>
+              {isEditing ? (
+                <input
+                  type="email"
+                  {...formik.getFieldProps('email')}
+                  required
+                  className="editing-input"
+                />
+              ) : (
+                <span>{userProfile.email}</span>
+              )}
+            </label>
+            <label className="label">
+              <span className="color-gray">Date of Birth</span>
+              {isEditing ? (
+                <input
+                  type="date"
+                  {...formik.getFieldProps('dateOfBirth')}
+                  required
+                  className="editing-input"
+                />
+              ) : (
+                <span>{userProfile.dateOfBirth}</span>
+              )}
+            </label>
+            <label className="label">
+              <span className="color-gray">Registration Date</span>
+              <span>{userProfile.registrationDate}</span>
+            </label>
+            <label className="label">
+              <span className="color-gray">Last Login Timestamp</span>
```

Comment:
Timestamp - I'm not sure that this word need to user, maybe better name is 'Last login date'?

Reply:
I agree and will fix it